whitelist-authenticator/src/main/java/de/intension/authentication/WhitelistAuthenticator.java

```diff
@@ -55,11 +54,14 @@ public void authenticate(AuthenticationFlowContext context)
      */
     private String getProviderIdFromContext(AuthenticationFlowContext context)
     {
-        String providerId = context.getUriInfo().getQueryParameters().getFirst(AdapterConstants.KC_IDP_HINT);
+        String idpHintName = context.getAuthenticatorConfig().getConfig()
+            .get(WhitelistConstants.IDP_HINT_PARAM_NAME);
+        String providerId = context.getUriInfo().getQueryParameters().getFirst(idpHintName);
```

Comment:
Maybe we should check, whether the idpHintName is empty or null, because the GUI does not check this.

Reply:
default is `kc_idp_hint` - so someone would have to edit the config and set it to empty on purpose - maybe I can add a constraint to the config parameter? :thinking:

Reply:
If it's possible, i think it would be a good thing. Up to you.

Reply:
I think it's impossible

Reply:
But I'll treat it as misconfiguration